src/mixins/helpers.js

```diff
@@ -43,6 +43,8 @@ var helpers = {
     var trackWidth = this.getWidth(ReactDOM.findDOMNode(this.refs.track));
     var slideWidth = this.getWidth(ReactDOM.findDOMNode(this))/props.slidesToShow;
 
+    var currentSlide = props.rtl ? slideCount - 1 - props.initialSlide : props.initialSlide;
```

Comment:
Why do you want to change currentSlide during update ?

May be need we need to check and update currentSlide only if slideCount < currentSlide